src/activities/ActivityUsageEntity.js

```diff
@@ -533,19 +546,19 @@ export class ActivityUsageEntity extends Entity {
 		const createNewGradeItem = scoreAndGrade.inGrades && !associatedGrade && scoreAndGrade.associateNewGradeAction;
 		const associateToExistingGrade = scoreAndGrade.inGrades && associatedGrade && associatedGrade.canAssociateGrade() && this.gradeHref() !== associatedGrade.href();
 		if (createNewGradeItem) {
```

Comment:
Definitely having a hard time getting my head around what is going on with all the grade and score stuff, but no surprise there. Hopefully if we can further rationalize how we are handling scores and grades and also add them to the activity usage in the future, we will be able to simplify this too.